apps/server/test/lexical/server/code_intelligence/entity_test.exs

```diff
@@ -299,30 +304,247 @@ defmodule Lexical.Server.CodeIntelligence.DefinitionTest do
+      code = ~q[
+        Some.Modul|e.
+        ]
+
+      assert {:error, _} = resolve(project, code)
+    end
+
+    test "returns an error if the cursor is not over an entity", %{project: project} do
+      code = ~q[
+        Beyond.The.End|
+        ]
+
+      assert {:error, :not_found} = resolve(project, code)
+
+      code = ~q[
+        | Beyond.The.End
+        ]
+
+      assert {:error, :not_found} = resolve(project, code)
+    end
+
+    test "returns module entities at and before the cursor", %{project: project} do
+      code = ~q[
+        At.The.En|d
+        ]
+
+      assert {:ok, {:module, At.The.End}} = resolve(project, code)
+
+      code = ~q[
+        Beginning.Of.The|.End
+        ]
+
+      assert {:ok, {:module, Beginning.Of.The}} = resolve(project, code)
+
+      code = ~q[
+        En|d.Of.The.Beginning
+        ]
+
+      assert {:ok, {:module, End}} = resolve(project, code)
+
+      code = ~q[
+        |At.The.Beginning
+        ]
+
+      assert {:ok, {:module, At}} = resolve(project, code)
+    end
+
+    # this test was added due to an off-by-one error that wasn't taking
+    # the separator into account when calculating offsets, causing the
+    # result to become more inaccurate as the alias grew longer
+    test "excludes the module segments after the hovered separator", %{project: project} do
+      code = ~q[
+        AAA|.BBB.CCC.DDD.EEE
+        ]
+
+      assert {:ok, {:module, AAA}} = resolve(project, code)
+
+      code = ~q[
+        AAA.BBB|.CCC.DDD.EEE
+        ]
+
+      assert {:ok, {:module, AAA.BBB}} = resolve(project, code)
+
+      code = ~q[
+        AAA.BBB.CCC|.DDD.EEE
+        ]
+
+      assert {:ok, {:module, AAA.BBB.CCC}} = resolve(project, code)
+
+      code = ~q[
+        AAA.BBB.CCC.DDD|.EEE
+        ]
+
+      assert {:ok, {:module, AAA.BBB.CCC.DDD}} = resolve(project, code)
+    end
+
+    test "returns module segments in nodes spanning multiple lines", %{project: project} do
+      code = ~q[
+        foo =
+          On.Another.Lin|e
+        ]
+
+      assert {:ok, {:module, On.Another.Line}} = resolve(project, code)
+
+      code = ~q[
+        On|.
+          Multiple.
+          Lines
+        ]
+
+      assert {:ok, {:module, On}} = resolve(project, code)
+
+      code = ~q[
+        On.
+          Multiple.
+          Line|s
+        ]
+
+      assert {:ok, {:module, On.Multiple.Lines}} = resolve(project, code)
+
+      code = ~q[
+        Enu|m.map(1..10, fn i ->
+          i + 1
+        end)
+        ]
+
+      assert {:ok, {:module, Enum}} = resolve(project, code)
+    end
+
+    test "resolves explicit aliases", %{project: project} do
+      code = ~q[
+        defmodule Example do
+          alias Long.Aliased.Module
+          Modul|e
+        end
+        ]
+
+      assert {:ok, {:module, Long.Aliased.Module}} = resolve(project, code)
+
+      code = ~q[
+        defmodule Example do
+          alias Long.Aliased.Module
+          Module.Neste|d
```

Comment:
Same problem, One example is sufficient; you are just duplicating some logic with the aliases tests. Moreover, I believe it is already addressed in the `aliases_tests.exs` file.

Reply:
Agreed that only one example is sufficient here.

I do not agree that we should consider the `aliases_test.exs` as sufficient, however, as there's no guarantee this API continues to use `Aliases` in the future. `Aliases` is a unit that should be tested, and this is a unit, one level of abstraction up, that also needs to be tested.